src/script/zcash_script.h

```diff
@@ -63,6 +63,26 @@ void* zcash_script_new_precomputed_tx(
     unsigned int txToLen,
     zcash_script_error* err);
 
+/// Deserializes the given transaction and precomputes values to improve
+/// script verification performance. Must be used for V5 transactions;
+/// may also be used for previous versions.
+///
+/// allPrevOutputs must point to the concatenation of all outputs from previous
+/// transactions that are spent by the inputs of the given transaction.
```

Comment:
So the intention is that we parse `txTo`, then use `txTo.vin.len()` as the counter for parsing `allPrevOutputs`? That seems reasonable; we'd load `allPrevOutputs` into a `CDataStream`, read `CTxOut` from it `txTo.vin.len()` times, and fail if we either run out of data, or there was excess data left over afterwards.

It would be _slightly_ cleaner on our side if the encoding were prefixed with a `CompactSize` denoting the number of `CTxOut`s (i.e. the exact same format as is used for `vout` in a transaction), because then we could just do:
```cpp
std::vector<CTxOut> allPrevOutputs;
try {
    ss >> allPrevOutputs;
} catch (std::ios_base::failure e) {
    // Fail
}
if (!ss.empty() || allPrevOutputs.len() != txTo.vin.len()) {
    // Fail
}
```
vs with the currently-proposed encoding (which requires inlining the parsing logic):
```cpp
std::vector<CTxOut> allPrevOutputs;
allPrevOutputs.resize(txTo.vin.len());
try {
    for (int i = 0; i < txTo.vin.len(); i++) {
        ss >> allPrevOutputs[i];
    }
} catch (std::ios_base::failure e) {
    // Fail
}
if (!ss.empty()) {
    // Fail
}
```

Reply:
Good idea, that also makes it easier for Zebra too since we can just serialize the vector. Changed in b459f197f